### PR TITLE
[TSK-241] configure gateway

### PR DESCRIPTION
**Pull request by Hoocki**

### CODE REVIEW

File: cloud/api-gateway/src/main/resources/application-local.yaml [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+          predicates:
+            - Path=/api/v1/notifications/**
+        - id: qr-generator
+          uri: http://localhost:8085
+          predicates:
+            - Path=/api/v1/qrcode/**
+        - id: auth-manager
+          uri: http://localhost:8086
+          predicates:
+            - Path=/api/v1/users/**,/api/v1/auth/**
```
Comment (adsukharev) [inline cloud/api-gateway/src/main/resources/application-local.yaml]:
```suggestion
            - Path=/api/v1/auth/users/**,/api/v1/auth/**
```
I don't like that for auth manager the path is `/api/v1/users/` .  Looks like it is for user-manager.

File: cloud/api-gateway/src/main/resources/application-local.yaml [hunk 2]
```diff
@@ -1,4 +1,28 @@
 server:
   port: 8080
   error:
-    include-message: always
\ No newline at end of file
+    include-message: always
+spring:
```
Comment (adsukharev) [inline cloud/api-gateway/src/main/resources/application-local.yaml]: move to prod config

File: cloud/api-gateway/src/main/resources/application-local.yaml [hunk 3]
```diff
@@ -1,4 +1,28 @@
...
+        - id: payment-manager
+          uri: http://localhost:8082
+          predicates:
+            - Path=/api/v1/wallets/**,/api/v1/payments/**
+        - id: notification-manager
+          uri: http://localhost:8083
+          predicates:
+            - Path=/api/v1/notifications/**
+        - id: qr-generator
+          uri: http://localhost:8085
```
Comment (adsukharev) [inline cloud/api-gateway/src/main/resources/application-local.yaml]: strange that you don't have 8084 port

**Review by adsukharev — COMMENTED** [review]
Did you check that it works?
I don't see main file

File: cloud/api-gateway/src/main/resources/application-local.yaml (hunk 3 above)
Comment (Hoocki) [reply cloud/api-gateway/src/main/resources/application-local.yaml]: 8084 - transaction-history

**Comment by Hoocki** [comment]
> Did you check that it works? I don't see main file

i checked, but i don't understand which main file is it about

**Comment by adsukharev** [comment]
> > Did you check that it works? I don't see main file
> 
> i checked, but i don't understand which main file is it about

I did not see ApiGatewayApplication. But you have already created it, no problem

**Comment by adsukharev** [comment]
add logger
```
logging:
  level:
    org.springframework.cloud.gateway: DEBUG
    reactor.netty.http.client: DEBUG
```

File: cloud/api-gateway/src/main/resources/application.yaml [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+            predicates:
+              - Path=/api/v1/notifications/**
+          - id: qr-generator
+            uri: http://localhost:8085
+            predicates:
+              - Path=/api/v1/qrcode/**
+          - id: auth-manager
+            uri: http://localhost:8086
+            predicates:
+              - Path=/api/v1/identity/users/**,/api/v1/auth/**
```
Comment (adsukharev) [inline cloud/api-gateway/src/main/resources/application.yaml]: Actually this path /api/v1/identity/users/** should not be in the API Gateway. This is for internal use only. We don't allow users directly send requests to authManager to create users. Same for notifications
Btw identity is not relevant name. Maybe `credentials` ?

File: cloud/api-gateway/src/main/resources/application.yaml [hunk 5]
```diff
@@ -1,3 +1,31 @@
...
+
+logging:
+  level:
+    org.springframework.cloud.gateway: DEBUG
+    reactor.netty.http.client: DEBUG
```
Comment (adsukharev) [inline cloud/api-gateway/src/main/resources/application.yaml]: debug logger is for local

File: cloud/api-gateway/src/main/resources/application.yaml [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+        routes:
+          - id: user-manager
+            uri: http://localhost:8081
+            predicates:
+              - Path=/api/v1/customers/**,/api/v1/merchants/**
+          - id: payment-manager
+            uri: http://localhost:8082
+            predicates:
+              - Path=/api/v1/wallets/**,/api/v1/payments/**
+          - id: notification-manager
```
Comment (adsukharev) [inline cloud/api-gateway/src/main/resources/application.yaml:15]: for internal use. Remove it

**Review by adsukharev — APPROVED** [review]

File: cloud/api-gateway/src/main/resources/application.yaml (hunk 6 above)
Comment (Hoocki) [reply cloud/api-gateway/src/main/resources/application.yaml:15]: remove notfication service or paths in payment-manager?

Comment (Hoocki) [reply cloud/api-gateway/src/main/resources/application.yaml:15]: i thought these paths should be able for user to get wallet and provide payment and so on

Comment (adsukharev) [reply cloud/api-gateway/src/main/resources/application.yaml:15]: Users cannot directly send requests to notification manager.

**Pull request merged**